cpp/tests/io/parquet_test.cpp

```diff
@@ -396,21 +396,19 @@ TYPED_TEST(ParquetWriterNumericTypeTest, SingleColumn)
   constexpr auto num_rows = 800;
   column_wrapper<TypeParam> col(sequence, sequence + num_rows, validity);
 
-  std::vector<std::unique_ptr<column>> cols;
-  cols.push_back(col.release());
-  auto expected = std::make_unique<table>(std::move(cols));
-  EXPECT_EQ(1, expected->num_columns());
+  auto expected = table_view{{col}};
+  EXPECT_EQ(1, expected.num_columns());
```

Comment:
Not really a part of the review, just taking the opportunity to mention this. This is a weird check, right? IMO it's testing `table`(previously)/`table_view`(now) constructor, rather then anything in Parquet. I would be happier without these asserts. Thoughts?

Reply:
I thought they were generally in here to prevent someone coming and adding a second column and not updating the rest of the test. It is an odd thing to see though, because why would you simply add a column without even looking at the rest of the test? I'd be happy to pull them out in this PR if you desire. Not much work and not too much point.

Reply:
Probably better to do it in a different PR, this one looks good to go:)